### DISCUSSION TITLE
dragging window title bar to center for two thirds sizing

### DISCUSSION THREAD

**Discussion by myusrn** [rxhanson/Rectangle#1128] [Q&A] [answered]
I just applied an install of rectangle v0.67 on macos ventura 13.2.1 running on macbook pro 13" m1 device.  

I'm trying to make the default "Thirds, drag towards center for two thirds" gesture work and i'm not getting any results. 

Is there a common new user mistake one makes to get this window placement and sizing gesture to work?

**Comment by rxhanson** [comment, marked as answer]
I'm planning on making some tutorial videos, starting this weekend. 

If you drag the window to the snap area and have a 1/3 footprint, you just continue to drag along the edge of the display toward the center, and the footprint should switch to 2/3. 

Let me know if that doesn't make sense and I'll start with a video demonstrating this, or if you think you're doing it correctly and something's not working then feel free to make a quick screen recording.

**Comment by myusrn** [comment]
Thanks yes i see it working now. My confusion was that i thought i had to drag the window to the actual center of the screen to trigger the snap not the bottom center.

**Comment by myusrn** [comment]
Status update.  

With the default setting of "Thirds, drag towards center for two thirds" set for the drag window to bottom center of screen i'm getting what looks like a little less than 1/3rd of the screen sized window now the suggested 2/3rds of the screen. 

If i change that setting to "Center Half" and do the drag window title to bottom center i'm getting a 1/2 the screen real estate sized window that is definitely bigger than the "Thirds, drag towards center for two thirds" result.

Now that it seems i have the gesture worked out for that drag to place and size window behavior am i still overlooking something to get it to size it to 2/3rds of the display real estate vs what looks like the 1/3rd of the display size i'm getting?

**Reply by rxhanson** [reply]
If I understand correctly, you just need to start on either the left or right 1/3, then drag it to the center third to see the 2/3 footprint. It sounds like you are just going straight to the center 1/3?

**Reply by myusrn** [reply]
thanks for the follow up and additional pointers to try and determine what i'm doing wrong about the required gesture. 

"you just need to start on either the left or right 1/3" . . . i interpret this to mean click on left or right 1/3rd of the title bar on window i want to move to center 2/3rds size and placement.

"then drag it to the center third" . . . i interpret this to mean to the center third of the screen down at the bottom just by the task bar

given those interpretations i still end up with the window placement and size doing what looks like little less than 1/3rd of screen size not center 2/3rds. conversely if i grab a window title bar, any part of it, and drag down to the left or right 1/3rd of screen and then move towards the center of the screen i do get a 2/3rds display sized window but the placement is the left or right 2/3rds of display not the center 2/3rds of the display.  maybe that is the expected result?

**Reply by rxhanson** [reply]
> but the placement is the left or right 2/3rds of display not the center 2/3rds of the display. maybe that is the expected result?

Indeed this is the expected result. Sorry for the confusion. At the moment, this snap area does not support center 2/3. I'll give this some thought and see if it makes sense to add this in as another option. I would like to add some more options, but I don't use this behavior so I haven't really given it much thought about more "compound snap areas" to add in. I'm open to suggestions for what anyone might want, or don't mind merging in a good pull request with additional compound snap areas.

**Reply by myusrn** [reply]
Thanks for confirmation that i now understand that gesture and expected result and can see why you'd want it. I regularly do development with idea using 2/3rds and debugging UI session in a remaining third or live meeting participation and note taking app. 

Currently i took your "Center" and changed ctrl+opt+C to ctrl+opt+M[iddle] and then mapped ctrl+opt+C[enter] to the "Center Half" option that had no mapping by default.  

I like center half [ and 1/3rd ] and use that placement and sizing all the time on when i'm reading documents on laptop or desktop 16:10 displays.  

I can see how there is many times where center half is more desirable than center 2/3rds where it ends up taking up so much of the display the difference between that and ctrl+opt+enter maximized window is not significant enough to warrant it.

**Comment by myusrn** [comment]
Any thoughts on whether or not making a center 2/3rds or 60% of screen option in addition to center 50% option is something a large portion of the user base would want vs having to use the drag to lower left/right that towards center to get a 2/3rds sized window placed on left or right side followed by use of center keyboard shortcut to move it to middle of screen?

**Reply by rxhanson** [reply]
I don’t have plans to add it in myself but see no reason to turn down a good pull request that adds in this option.

**Reply by myusrn** [reply]
Haven't written any macos platform apps before using swift in xcode ide. If i can carve out the time to have a look at what it would take to introduce a center 60 - 66% of screen option will submit a pull request.

**Reply by rxhanson** [reply]
I re-read your comment and realized that I think I misunderstood. I thought you wanted a new compound drag-to-snap area that would result in center 2/3 rather than side 2/3. 

If you just want a keyboard shortcut for it, you can use this terminal command:
https://github.com/rxhanson/Rectangle/blob/master/TerminalCommands.md#add-an-extra-centering-command-with-custom-size

Alternatively, you could use [Rectangle Pro](https://rectangleapp.com/pro) (paid) that has center 2/3 or a shortcut builder for any size/position. 

This is one of those edge cases with Rectangle, where in order to maintain simplicity, I had to draw the line on how many actions get added into the UI. If anyone really wants to add in center 2/3, I'd still be fine merging it in, but only as a terminal command config and not in the UI, just like that "extra centering one" with the terminal command.

**Reply by myusrn** [reply]
Thanks for those clarifications and additional details. I can appreciate how you don't want the UI littered with every possible scenario only the mainstream use cases. Looked at url#bookmark provided for creating a keyboard shortcut [ e.g. opt/alt+c ] that would do a center 2/3rds placement and sizing action.  Wasn't clear how first of the three terminal command lines ties in with the next two command lines that specify the height and width settings to go with the keyboard shortcut specified in the first statements -float switch.  Also how would you flush that keyboard shortcut if you later decided to change it.

**Reply by rxhanson** [reply]
The first command is the keyboard shortcut (key & modifier flags). 

There's no "tying" it that you have to do; it's three separate entries for that specific shortcut.

To flush it, do
`defaults delete com.knollsoft.Rectangle specified`

and restart the app.

**Reply by myusrn** [reply]
thanks that worked for allowing me to tie opt[/alt]+c to the center 60% [ or 66.67% 2/3rds ] placement and size on a qhd 2560x1440 desktop display using the following commands. that said i can see how the [Rectangle Pro](https://rectangleapp.com/pro) out of the box solution for this would be desirable. 

see any issues in your experience with tying up opt[/alt]+c keyboard shortcut in terms of stomping on some other system or mainstream application use of it by default?

```
defaults write com.knollsoft.Rectangle specified -dict-add keyCode -float 8 modifierFlags -float 524576
defaults write com.knollsoft.Rectangle specifiedHeight -float 1440
defaults write com.knollsoft.Rectangle specifiedWidth -float 1536 [ or 1706 ]
restart rectangle app   
   
# and confirmed following unwinds that keyboard shortcut setup  

defaults delete com.knollsoft.Rectangle specified   
restart rectangle app  
```
